controller/eks-cluster-config-handler.go

```diff
@@ -1185,14 +1189,25 @@ func (h *Handler) updateUpstreamClusterState(upstreamSpec *eksv1.EKSClusterConfi
 			ClusterName:   aws.String(config.Spec.DisplayName),
 		}
 
+		// there exists a launch template for this node group
 		if upstreamNg.LaunchTemplate != nil {
 			upstreamTemplateVersion := aws.Int64Value(upstreamNg.LaunchTemplate.Version)
 			var err error
 			lt := ng.LaunchTemplate
 
+			// once the cluster is created, update the node group launch template with the default cluster security
+			// group generated by EKS
+			addDefaultClusterSecurityGroup := false
+			if !reflect.DeepEqual(config.Status.SecurityGroups, upstreamSpec.SecurityGroups) {
```

Comment:
Would it be possible for these to have equal elements but in different orders?

Reply:
Not in the case of a provisioned EKS cluster. Once rancher defined security groups are set, they are pulled in that same order from upstream and the cluster default security group is always appended to the _end_ of that list. It will always have the order [<rancher sgs (in order selected in the rancher UI>, <default sg>]. Once user defined security groups are selected, they cannot be modified